src/utils/uint256.cairo

```diff
@@ -102,6 +104,192 @@ func uint256_sub{range_check_ptr}(a: Uint256, b: Uint256) -> (res: Uint256) {
     }
 }
 
+// ! The following functions are taken from starkware's cairo common library
+// ! to use the optimized uint256_add and uint256_sub, with inlined uint256_check
+
+// Returns 1 if the first signed integer is less than the second signed integer.
+func uint256_signed_lt{range_check_ptr}(a: Uint256, b: Uint256) -> (res: felt) {
+    alloc_locals;
+    let (a, _) = uint256_add(a, cast((low=0, high=2 ** 127), Uint256));
+    let (b, _) = uint256_add(b, cast((low=0, high=2 ** 127), Uint256));
```

Comment:
`cast((low=0, high=2 ** 127), Uint256)` is `Uint256(low=0, high=2 ** 127)`

Reply:
this came from the core library directly

Reply:
maybe inherited from previous languages versions 🤷